usertask-event-listener.js

```diff
@@ -0,0 +1,168 @@
+                                    flowNodeInstanceId: userTaskNotification.flowNodeInstanceId,
+                                    ...query,
+                                };
+
+                                const matchingFlowNodes = await client.userTasks.query(newQuery, {
+                                    identity: currentIdentity,
+                                });
+
+                                if (matchingFlowNodes.userTasks && matchingFlowNodes.userTasks.length == 1) {
+                                    const userTask = matchingFlowNodes.userTasks[0];
+
+                                    node.send({
+                                        payload: {
+                                            flowNodeInstanceId: userTaskNotification.flowNodeInstanceId,
+                                            userTaskEvent: userTaskNotification,
+                                            userTask: userTask,
+                                            action: 'reserved',
+                                            type: 'usertask',
+                                        },
+                                    });
+                                }
+                            },
+                            { identity: currentIdentity }
+                        );
+                    case 'reservation-canceled':
+                        return await client.userTasks.onUserTaskReservationCanceled(
+                            async (userTaskNotification) => {
+                                if (config.usertask != '' && config.usertask != userTaskNotification.flowNodeId) return;
+                                const newQuery = {
+                                    flowNodeInstanceId: userTaskNotification.flowNodeInstanceId,
+                                    ...query,
+                                };
+
+                                const matchingFlowNodes = await client.userTasks.query(newQuery, {
+                                    identity: currentIdentity,
+                                });
+
+                                if (matchingFlowNodes.userTasks && matchingFlowNodes.userTasks.length == 1) {
+                                    const userTask = matchingFlowNodes.userTasks[0];
+
+                                    node.send({
+                                        payload: {
+                                            flowNodeInstanceId: userTaskNotification.flowNodeInstanceId,
+                                            userTaskEvent: userTaskNotification,
+                                            userTask: userTask,
+                                            action: 'reservation-canceled',
+                                            type: 'usertask',
+                                        },
+                                    });
+                                }
+                            },
+                            { identity: currentIdentity }
+                        );
```

Comment:
unterscheiden sich die ganze cases nicht nur in der action?

Könnte man nicht eine Methode rausziehen?

Reply:
Ich war mir nicht sicher was hier mehr angebracht ist: LoB oder CleanCode. Aber du hast recht ist schon sehr viel Wiederholung drinnen gewesen. Ist mir beim Coden nicht so aufgefallen :)